### PR TITLE
WDPD-163 Basic Ratepay Invoice

### PR DESCRIPTION
**Pull request by ps-cgrach**
### This PR

* adds a basic setup for the Ratepay Invoice payment method

### Notes

* this is just the basic configuration to get the actual implementation of this payment method started

### Jira Links

* https://jira.parkside.at/browse/WDPD-163

### CODE REVIEW

File: Model/RatepayInvoicePaymentMethod.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ * @since 1.2.0
+ */
+class RatepayInvoicePaymentMethod extends PaymentMethod
+{
+    /**
+     * @inheritdoc
+     *
+     * @since 1.2.0
+     */
+    protected static $_sName = "ratepay-invoice";
```
Comment (ps-holler) [inline Model/RatepayInvoicePaymentMethod.php]: So gar we always skipped delimiters in payment names, e.g. "creditcard" or "sepacredit". So maybe we should just call this "ratepayinvoice"?

File: Model/RatepayInvoicePaymentMethod.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    public function getTransaction()
+    {
+        return new RatepayInvoiceTransaction();
+    }
+
+    public function getConfigFields()
+    {
+        $aAdditionalFields = [
+            'descriptor' => [
+                'type'        => 'select',
```
Comment (ps-holler) [inline Model/RatepayInvoicePaymentMethod.php]: Inconsistent use of spaces.

File: Model/RatepayInvoicePaymentMethod.php (hunk 1 above)
Comment (ps-cgrach) [reply Model/RatepayInvoicePaymentMethod.php]: no, this is the name we need for the transaction service.

File: default_payment_config.xml [hunk 3]
```diff
@@ -86,6 +86,23 @@
         <wdoxidee_countrycode>en_gb</wdoxidee_countrycode>
         <wdoxidee_logovariant>standard</wdoxidee_logovariant>
     </payment>
+    <payment>
+      <oxid>wdratepay-invoice</oxid>
+      <oxactive>0</oxactive>
+      <oxtoamount>1000000</oxtoamount>
+      <oxdesc>Wirecard Garantierter Kauf auf Rechnung</oxdesc>
+      <oxdesc_1>Wirecard Guaranteed Invoice by Wirecard</oxdesc_1>
```
Comment (ps-dfritzsch) [inline default_payment_config.xml]: Redundant use of "Wirecard"

Comment (ps-cgrach) [reply default_payment_config.xml]: that string is from phraseapp

File: Model/RatepayInvoicePaymentMethod.php [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    public function getPublicFieldNames()
+    {
+        return array_merge(
+            parent::getPublicFieldNames(),
+            ['additionalInfo', 'deleteCanceledOrder', 'deleteFailedOrder']
```
Comment (ps-amaureder) [inline Model/RatepayInvoicePaymentMethod.php]: is there a reason, why we do not have descriptor in here?

Comment (ps-mstifter) [reply Model/RatepayInvoicePaymentMethod.php]: There is not, added it 👍

File: Model/RatepayInvoicePaymentMethod.php (hunk 1 above)
Comment (tomazpu) [reply Model/RatepayInvoicePaymentMethod.php]: You could use the consts from the transaction. You don't need to change it. Just to maybe think of it if its better.

File: Model/RatepayInvoicePaymentMethod.php [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    public function getConfigFields()
```
Comment (tomazpu) [inline Model/RatepayInvoicePaymentMethod.php]: Im missing, config fields for configuring the allowed currencies, shipping, billing countries, if billing_shipping needs to be the same, min/max basket amount

File: default_payment_config.xml [hunk 6]
```diff
@@ -86,11 +90,30 @@
...
         <wdoxidee_logovariant>standard</wdoxidee_logovariant>
     </payment>
+    <payment>
+      <oxid>wdratepay-invoice</oxid>
+      <oxactive>0</oxactive>
+      <oxfromamount>20</oxfromamount>
+      <oxtoamount>3500</oxtoamount>
+      <oxdesc>Wirecard Garantierter Kauf auf Rechnung</oxdesc>
+      <oxdesc_1>Wirecard Guaranteed Invoice by Wirecard</oxdesc_1>
+      <oxsort>55</oxsort>
```
Comment (tomazpu) [inline default_payment_config.xml]: same here im missing config fields

File: Model/RatepayInvoicePaymentMethod.php (hunk 5 above)
Comment (ps-mstifter) [reply Model/RatepayInvoicePaymentMethod.php]: As mentioned in the PR description we only added the basic configuration for this payment method and intentionally left the currencies and shipping, billing address settings out for now to have a common ground to get started with the development.

Min/max basket amount can be handled automatically through OXID's payment method configuration.

File: default_payment_config.xml (hunk 6 above)
Comment (ps-mstifter) [reply default_payment_config.xml]: same as above

File: Model/RatepayInvoicePaymentMethod.php (hunk 1 above)
Comment (ps-mstifter) [reply Model/RatepayInvoicePaymentMethod.php]: Thanks for the hint, we will consider that 👍

**Pull request merged**